ipaplatform/base/services.py

```diff
@@ -298,14 +302,25 @@ def start(self, instance_name="", capture_output=True, wait=True):
             instance_name,
             update_service_list=update_service_list)
 
-    def restart(self, instance_name="", capture_output=True, wait=True):
-        ipautil.run([paths.SYSTEMCTL, "restart",
-                     self.service_instance(instance_name)],
+    def _restart_base(self, instance_name, operation, capture_output=True,
```

Comment:
It was meant as a base class for: "reload", "restart", "try-restart", "reload-or-restart" and "try-reload-or-restart". The name could be `reload_restart_base`, but given that it is private/protect function then it can be renamed at any time in the future without any trouble when new methods are introduced as needed..